src/core/datatypes/forte_string_fixed.h

```diff
@@ -69,18 +70,26 @@ class CIEC_STRING_FIXED final : public CIEC_STRING {
       return *this;
     }
 
-    CIEC_STRING_FIXED &operator=(const CIEC_STRING_FIXED &paValue) {
-      getStorageMutable() = paValue.getStorage();
+    template<size_t maxLengthSrc>
+    CIEC_STRING_FIXED &operator=(const CIEC_STRING_FIXED<maxLengthSrc> &paValue) {
+      if constexpr (maxLengthSrc > maxLength) {
+        const std::string_view stringView(paValue.getStorage());
+        getStorageMutable() = stringView.substr(0, maxLength);
+      } else {
+        getStorageMutable() = paValue.getStorage();
+      }
       return *this;
     }
 
-    CIEC_STRING_FIXED &operator=(CIEC_STRING_FIXED &&paValue) {
+    template<size_t maxLengthSrc>
+    CIEC_STRING_FIXED &operator=(CIEC_STRING_FIXED<maxLengthSrc> &&paValue) {
       getStorageMutable() = std::move(paValue.getStorageMutable());
+      if constexpr (maxLengthSrc > maxLength) {
+        getStorageMutable().resize(std::min<std::size_t>(this->length(), maxLength));
+      }
       return *this;
     }
```

Comment:
Am I correct that this is necessary for MSVC but does not add new types of operator=?

Reply:
The commit included an improvement, changing the definition order is enough for now. I removed the improvements from this PR, maybe we can merge it later.